### PR TITLE
Change `UiScale` to a tuple struct

### PR DESCRIPTION
**Pull request by st0rmbtw**
# Objective

Inconvenient initialization of `UiScale`

## Solution

Change `UiScale` to a tuple struct

## Migration Guide

Replace initialization of `UiScale` like ```UiScale { scale: 1.0 }``` with ```UiScale(1.0)```

### CODE REVIEW

File: crates/bevy_ui/src/lib.rs [hunk 1]
```diff
@@ -73,6 +73,12 @@ pub struct UiScale {
     pub scale: f64,
 }
 
+impl UiScale {
+    pub const fn new(scale: f64) -> Self {
```
Comment (alice-i-cecile) [inline crates/bevy_ui/src/lib.rs:77]: Missing docs :) Very simple is fine.

**Comment by alice-i-cecile** [comment]
Nice and simple, but I'm not fully sold. Can you say a bit more about why you prefer this syntax?

**Comment by st0rmbtw** [comment]
It feels more consistent and looks nicer for me :)

**Comment by mockersf** [comment]
Would it be simpler to change the struct to a tuple struct? `struct UiScale(f32);`

**Comment by st0rmbtw** [comment]
I think so, but wouldn't it be a breaking change?

**Comment by alice-i-cecile** [comment]
Yep, but breaking changes are totally fine in Bevy at this stage. Especially for a very niche part of the API like this.

I'd prefer a tuple struct over the current change as well, although I'm neutral on tuple vs named struct.

**Comment by st0rmbtw** [comment]
Done

**Comment by mockersf** [comment]
I prefer named struct when the name brings new information. I don't think a single field `scale` on a `UiScale` struct brings much...

You can probably derive `Deref` and `DerefMut` for that struct, and you'll need to fix all the places it's used...

**Comment by st0rmbtw** [comment]
@mockersf Should I add a method like `get` to return a value of UiScale or just use .0 syntax?

**Review by ickshonpe — APPROVED** [review]
I prefer the tuple struct too.

**Review by ickshonpe — COMMENTED** [review]
The UiScale doc comment isn't very informative atm. Maybe it could be changed to say something about how logical coordinates are multiplied by window scale factor and UiScale to get physical coordinates.

**Review by tguichaoua — APPROVED** [review]
IMHO tuple structs are better for wrapping a single meaningless value.
Also here it avoids the repetition of the word "scale".

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**